src/server/Agent.js

```diff
@@ -61,17 +65,26 @@ export default class Agent extends SideAgent {
     _startListening() {
         this.server = asyncNet.createServer();
 
-        this.server.on('connection', userSocket => this._startServing(userSocket));
+        this.server.on('connection', userSocket =>  {
+            if (!this._serverSocket) {
+                this._serverSocket = userSocket;
+                this._startServing();
+                this._serverClosed = this.server.closeAsync();
+            } else { // get here only if closeAsync() has not yet made server stop listening
+                console.log("does not support more than one server connection");
+                userSocket.destroy();
+            }
+        });
 
         const addr = Gadgets.FinalizeListeningAddress(this.address());
-        return this.server.listenAsync(addr.port, addr.host).
+        return this.server.listenAsync(addr.port, addr.host);
             bind(this).
             tap(this._startedListening);
```

Comment:
Something went wrong here. The original code was binding `this` to the listenAsync() event and then calling _startedListening. Or, at least, that was the intent. The current trailing code looks unreachable. Do we still call _startedListenting() somewhere?

Reply:
I'll undo this accidental/unintentional change. I missed it probably because the unreached _startListening() essentially only prints a debug line.